src/metabase/native_query_analyzer.clj

```diff
@@ -163,7 +163,7 @@
   query. Currently only support SQL-based dialects."
   [query]
   (when (and (active?) (:native query))
-    (let [driver (driver.u/database->driver (:database query))]
+    (let [driver (lib.native/engine query)]
```

Comment:
It turns out we can't actually make this change (yet?) as our test queries don't have the required `:lib/type` and `:stages` values used by the validation for this helper.